vsintegration/src/FSharp.Editor/Refactor/ChangeDerefToValueRefactoring.fs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Composition
+open System.Threading
+
+open FSharp.Compiler
+open FSharp.Compiler.CodeAnalysis
+open FSharp.Compiler.Symbols
+open FSharp.Compiler.Text
+open FSharp.Compiler.Syntax
+
+open Microsoft.CodeAnalysis.Text
+open Microsoft.CodeAnalysis.CodeRefactorings
+open Microsoft.CodeAnalysis.CodeActions
+
+[<ExportCodeRefactoringProvider(FSharpConstants.FSharpLanguageName, Name = "AddExplicitTypeToParameter"); Shared>]
+type internal FSharpAddExplicitTypeToParameterRefactoring
```

Comment:
is this classname/refactoring provider name correct? this feels more like a `ReplaceDereferenceOpWithValueMember` than it does a `AddExplicitTypeToParameter`.

Reply:
whoops

Reply:
it's 'a some copy pasta